ambry-store/src/main/java/com/github/ambry/store/RemoteTokenTracker.java

```diff
@@ -69,7 +103,204 @@ void refreshPeerReplicaTokens() {
   /**
    * @return a snapshot of peer replica to token map.
    */
-  Map<String, FindToken> getPeerReplicaAndToken() {
+  Map<String, Pair<Long, FindToken>> getPeerReplicaAndToken() {
     return new HashMap<>(peerReplicaAndToken);
   }
+
+  /**
+   * @return the file path which persists the remote token.
+   */
+  public String getPersistFilePath() {
+    return localReplica.getReplicaPath() + File.separator + REMOTE_TOKEN_FILE_NAME;
+  }
+
+  /**
+   * Provide a way to explicitly persist the token besides the background persistor
+   */
+  public boolean persistToken() {
+    try {
+      persistor.persist();
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Disable this {@link RemoteTokenTracker} and cancel the scheduled persistor.
+   */
+  @Override
+  public void close() {
+    persistor.cancel();
+  }
+
+  /**
+   * Start the background persistor.
+   */
+  public void start(boolean enablePersist) {
+    if (enablePersist) {
+      if (this.scheduler != null) {
+        this.scheduler.scheduleAtFixedRate(persistor, 0, REMOTE_TOKEN_FILE_PERSIST_INTERVAL_IN_SECONDS,
```

Comment:
we can have a configuration for the interval. As a matter of fact, we can use this value to enable the persistence. If this value is 0 or negative, we don't enable the persistence, but if it's not, we enable it.

Reply:
anyway, just think we should probably have a configuration for this.

Reply:
Changed.